src/datasets/domain/models/Dataset.ts

```diff
@@ -0,0 +1,40 @@
+export interface Dataset {
+  id: number;
+  persistentId: string;
+  versionId: number;
+  versionInfo: DatasetVersionInfo;
+  license: DatasetLicense;
+  metadataBlocks: DatasetMetadataBlock[];
+}
+
+export interface DatasetVersionInfo {
+  majorNumber: number;
+  minorNumber: number;
+  state: DatasetVersionState;
+  createTime: Date;
+  lastUpdateTime: Date;
+  releaseTime?: Date;
+}
+
+export enum DatasetVersionState {
+  DRAFT = 'DRAFT',
+  RELEASED = 'RELEASED',
+  ARCHIVED = 'ARCHIVED',
+  DEACCESSIONED = 'DEACCESSIONED',
+}
```

Comment:
I found some more states when working on the UI Dataset labels, should we add them here? Or should I get them from other data of the dataset

```suggestion
export enum DatasetVersionState {
  DRAFT = 'DRAFT',
  RELEASED = 'RELEASED',
  ARCHIVED = 'ARCHIVED',
  DEACCESSIONED = 'DEACCESSIONED',
  EMBARGOED = 'EMBARGOED',
  IN_REVIEW = 'IN REVIEW'
}
```

Reply:
The states you mention are never returned from the Datasets API. The API always returns an enum value of the VersionState enum, which does not contain the states you mention. See: https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/DatasetVersion.java#L101

At the moment I'm not sure if the states you mention can be inferred from other endpoint payloads, and if it makes sense to set them as states at the same level as the other states from VersionState enum, but at the moment I would not include those states in any model, domain or UI.

Reply:
FWIW: I think In review is based on whether the dataset has an In review lock (checkable via the get locks API). Embargoed is dynamic - true until the embargo end dates for files pass. I'm not sure whether its directly accessible now via API (versus having to calculate from the file embargo info).

Reply:
Ok, @GPortas do we create an issue to handle the other states? because we are going to need them in the UI

Reply:
@MellyGray Sure https://github.com/IQSS/dataverse-client-javascript/issues/62